src/gui/osutils/winutils/WinUtils.cpp

```diff
@@ -142,6 +142,34 @@ bool WinUtils::isHighContrastMode() const
     return (settings.value("Flags").toInt() & 1u) != 0;
 }
 
+QString WinUtils::getDefaultApplicationForUrl(const QUrl& url)
```

Comment:
There is no need for this incorrect registry handling, just call `AssocQueryStringW(ASSOCF_NOTRUNCATE | ASSOCF_INIT_IGNOREUNKNOWN, ASSOCSTR_EXECUTABLE, L"http", 0, ...)`